### DISCUSSION TITLE
Can I switch between multiple instances of pnpm?

### DISCUSSION THREAD

**Discussion by RoyRao2333** [pnpm/pnpm#6322] [Q&A] [answered]
I would like to try pnpm 8 in my new projects, but my team's old projects only support pnpm@^7.

Can I install both pnpm@7.30.5 and pnpm@8, then switch between them in current project or current terminal session?

Any ideas would be appreciated. :)

**Comment by RexSkz** [comment, marked as answer]
The `pnpm dlx` (fetches a package from the registry without installing it as a dependency, hotloads it, and runs whatever default command binary it exposes) should work:

```bash
$ pnpm --package=pnpm@8 dlx pnpm -v
8.1.0
```

Adding an alias makes it more convenient:

```bash
$ alias pnpm8="pnpm --package=pnpm@8 dlx pnpm"
$ pnpm8 -v
8.1.0
```

For more information about `pnpm dlx`, please refer to: https://pnpm.io/cli/dlx

**Reply by zkochan** [reply]
Interesting solution, though it might be quite slow.

**Reply by RoyRao2333** [reply]
@zkochan 

Yes, it might. Still a pretty good practice though.

So is there any official solutions for now? 👀

**Reply by zkochan** [reply]
you may use corepack

**Reply by RexSkz** [reply]
It's cool that Node.js provides an official solution for package managers. It's experimental and needs to be enabled manually using `corepack enable`. Hope it becomes formal :)

**Comment by RoyRao2333** [comment]
@RexSkz 

Thanks for the reply! It looks great and I'll try it right away. 🤟🏻

By the way, what's the usage of `--package` option? Looks like it chooses a certain version of `pnpm` to execute but I don't see this option in official docs anywhere...

**Reply by RexSkz** [reply]
It's here~ https://pnpm.io/cli/dlx#--package-name

**Reply by RoyRao2333** [reply]
@RexSkz And yes, this command actually works! Thanks for saving my day! 🎉
